validations/scripts/utils/errors.py

```diff
@@ -0,0 +1,201 @@
+    pass
+
+
+class ProvisionerCliError(ProvisionerError):
+    pass
+
+
+
+class UnknownParamError(ProvisionerError):
+    pass
+
+
+class LogMsgTooLong(ProvisionerError):
+    pass
+
+
+
+class SubprocessCmdError(ProvisionerError):
+    _prvsnr_type_ = True
+
+    def __init__(
+        self, cmd, cmd_args: Any, reason: str = 'unknown'
+    ):
+        self.cmd = cmd
+        self.cmd_args = cmd_args
+        self.reason = reason
+
+    def __str__(self):
+        return (
+            "subprocess command failed, reason {}, args {}"
+            .format(self.reason, self.cmd_args)
+        )
+
+class PrvsnrTypeDecodeError(ProvisionerError, ValueError):
+    _prvsnr_type_ = True
+
+    def __init__(self, spec: Dict, reason: Union[str, Exception]):
+        self.spec = spec
+        self.reason = reason
+
+    def __str__(self):
+        return (
+            'decode failed for {}, reason: {!r}'
+            .format(self.spec, self.reason)
+        )
+
+
+class SWUpdateRepoSourceError(ProvisionerError, ValueError):
+    _prvsnr_type_ = True
+
+    def __init__(self, source: str, reason: str):
+        self.source = source
+        self.reason = reason
+
+    def __str__(self):
+        return (
+            'repo source {} is not acceptable, reason: {!r}'
+            .format(self.source, self.reason)
+        )
+
+
+class PrvsnrCmdError(ProvisionerError):
+    def __init__(self, cmd_id: str):
+        self.cmd_id = cmd_id
+
+
+class PrvsnrCmdNotFoundError(ProvisionerError):
+    pass
+
+class PrvsnrCmdNotFinishedError(ProvisionerError):
+    pass
+
+
+class ClusterMaintenanceError(ProvisionerError):
+    _prvsnr_type_ = True
+
+    def __init__(self, enable: bool, reason: Union[Exception, str]):
+        self.enable = enable
+        self.reason = reason
+
+    def __str__(self):
+        return (
+            'failed to {} cluster maintenance, reason: {!r}'
+            .format('enable' if self.enable else 'disable', self.reason)
+        )
+
+
+class ClusterMaintenanceEnableError(ClusterMaintenanceError):
+    def __init__(self, reason: Union[Exception, str]):
+        super().__init__(True, reason)
+
+
+class ClusterMaintenanceDisableError(ClusterMaintenanceError):
+    def __init__(self, reason: Union[Exception, str]):
+        super().__init__(False, reason)
+
+class SWStackUpdateError(ProvisionerError):
+    _prvsnr_type_ = True
+
+    def __init__(self, reason: Union[Exception, str]):
+        self.reason = reason
+
+    def __str__(self):
+        return (
+            'failed to update SW stack, reason: {!r}'
+            .format(self.reason)
+        )
+
+
+class HAPostUpdateError(ProvisionerError):
+    _prvsnr_type_ = True
+
+    def __init__(self, reason: Union[Exception, str]):
+        self.reason = reason
+
+    def __str__(self):
+        return (
+            'failed to apply Hare post_update logic, reason: {!r}'
+            .format(self.reason)
+        )
+
+
+class ClusterNotHealthyError(ProvisionerError):
```

Comment:
If at all we are planning to implement this, it should be Python version of:
https://github.com/Seagate/cortx-prvsnr/blob/pre-cortx-1.0/cli/src/common_utils/utility_scripts.sh#L51

Also, no Provisioner specific implementation: `ProvisionerError`

Reply:
Hello @ypise , Thank you. 
Yes, like I had mentioned previously, I created the initial draft as a sample for how other files will look in Validations Framework. since we're continuously implementing new ideas and changes to the framework, I just had them as placeholders. I've updated it all to latest version in the today's commit. Some functions are yet to be tested and changed accordingly. But this latest commit gives an overall idea on the complete structure of the framework.